tests/unit/test_download.py

```diff
@@ -490,18 +490,53 @@ def test_insecure_host_cache_is_not_enabled(self, tmpdir):
         assert not hasattr(session.adapters["https://example.com/"], "cache")
 
 
-def test_get_credentials():
+@pytest.mark.parametrize(["input_url", "url", "username", "password"], [
+    (
+        "http://user@email.com:password@example.com/path",
```

Comment:
I thought we were requiring the `@` to be encoded?

Reply:
Yep yep - I got confused at some point. 🙈